### PR TITLE
add .end([opts], cb)

### PR DESCRIPTION
**Pull request by juliangruber**

### CODE REVIEW

**Comment by juliangruber** [comment]
this can be sped up by skipping the fstat call when both mtime and atime are passed, shall i add that?

File: index.js [hunk 1]
```diff
@@ -129,6 +129,29 @@ RandomAccessFile.prototype.close = function (cb) {
...
+
+  function onopen (err) {
+    if (err) return cb(err)
+    if (!opts.atime && !opts.mtime) return cb()
+
+    fs.fstat(self.fd, function (err, stat) {
+      if (err) return cb(err)
+
+      var atime = opts.atime || stat.atime
+      var mtime = opts.mtime || stat.mtime
```
Comment (mafintosh) [inline index.js]: could you make these specifiable in the instance constructor as well?

**Comment by mafintosh** [comment]
1.3.0

**Comment by feross** [comment]
Why is this method named `end`?

**Comment by mafintosh** [comment]
the method will be called when the entire file has been written. do you have a suggestion for a better name for making it more clear? i suggested .end because its like stream.end() ish

**Comment by mafintosh** [comment]
a use case i wanna support is renaming the file from a tmp file to the actual file name when file is completely written

**Comment by feross** [comment]
Nice. That's a good use-case to support. I think the name is fine too. But I don't understand how the code does for `end` does that. It looks like it just sets the file timestamps and calls the callback immediately. Am I misreading it?

**Comment by juliangruber** [comment]
You're not misreading it, it's simply not there yet. We needed the mtime stuff now so we started with the .end() method but this is by far not all that's going to happen there.

**Comment by feross** [comment]
Gotcha!

**Comment by juliangruber** [comment]
(adding docs pr)

**Pull request merged**